hq_superset/oauth.py

```diff
@@ -33,6 +36,53 @@ def set_oauth_session(self, provider, oauth_response):
         # }
         session["oauth_response"] = oauth_response
 
+    def ensure_domain_role_created(self, domain):
+        # This inbuilt method creates only if the role doesn't exist.
+        return self.add_role(get_role_name_for_domain(domain))
+
+    def ensure_schema_perm_created(self, domain):
+        menu_name = self.get_schema_perm(get_ucr_database(), domain)
+        permission = self.find_permission_view_menu("schema_access", menu_name)
+        if not permission:
+            permission = self.add_permission_view_menu("schema_access", menu_name)
+        return permission
+
+    def sync_domain_role(self, domain):
+        # This creates DB schema, role and schema permissions for the domain and
+        #   assigns the role to the current_user
+        # self.create_schema(domain)
+        permission = self.ensure_schema_perm_created(domain)
+        role = self.ensure_domain_role_created(domain)
+        role.permissions = [permission]
+        # Todo; may need to add the permission to admin role as well
+        self.get_session.add(role)
+        self.get_session.commit()
+        import pdb; pdb.set_trace()
```

Comment:
eep

Reply:
This is removed in the other PR https://github.com/dimagi/hq_superset/pull/7/files#